server/util/testing/flags/flags.go

```diff
@@ -3,18 +3,27 @@ package flags
 import (
 	"flag"
 	"testing"
+
+	"github.com/buildbuddy-io/buildbuddy/server/config"
 )
 
 // Set a flag value and register a cleanup function to restore the flag
 // to its original value after the given test is complete.
 func Set(t testing.TB, name, value string) {
+	config.RegisterAndParseFlags()
 	f := flag.Lookup(name)
 	if f == nil {
 		t.Fatalf("Undefined flag: %s", name)
 	}
 	original := f.Value.String()
+	originalSetMap := config.GetOriginalSetFlags()
```

Comment:
nit: it's bad hygiene to return a mutable structure owned by a different object and manipulate it outside of that object. 
i think it would be slightly better to have explicit `TestOnlySetFlag` and `TestOnlyUnsetFlag` methods to do this manipulation. `TestOnlySetFlag` could return a bool to tell you whether it's a new flag.